### PR TITLE
fix: Fix export events module(URL fetch logic)

### LINKED ISSUE #3293: Event Export non-functional
**Describe the bug**
Exporting an event isn't working due to wrong logic pertaining to URL fetching and downloads.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'Event Dashboard'
2. Export the event (Press start, wait... and download)
3. See error

**Expected behaviour**
Event Exports should work

**Screenshots**
Screenshot 2019-07-19 at 10 49 12 AM

### PR DESCRIPTION
**Pull request by mrsaicharan1**
EventExport
Fixes #3293 

#### Short description of what this resolves:
Fixes event exports with the help of loader service

Related Server fix:

#### Changes proposed in this pull request:

- Implemented an action to fix downloads & file serving
- Modified component template logic

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/events/view/export/download-zip.js [hunk 1]
```diff
@@ -1,4 +1,27 @@
 import Component from '@ember/component';
+import { action } from '@ember/object';
 
-export default Component.extend({
-});
+export default class extends Component {
+    @action
```
Comment (niranjan94) [inline app/components/events/view/export/download-zip.js]: Incorrect spacing

File: app/components/events/view/export/download-zip.js [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+  exportEventDownload(eventDownloadUrl) {
+    this.set('isLoading', true);
+    this.loader
```
Comment (niranjan94) [inline app/components/events/view/export/download-zip.js]: Use async-await

File: app/components/events/view/export/download-zip.js [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+        const anchor = document.createElement('a');
+        anchor.style.display = 'none';
+        anchor.href = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
+        anchor.download = 'EventExport.zip';
+        anchor.click();
+        this.notify.success(this.l10n.t('Exported Event Downloaded successfully.'));
+      })
+      .catch(e => {
+        console.warn(e);
+        this.notify.error(this.l10n.t('Unexpected error occurred.'));
```
Comment (niranjan94) [inline app/components/events/view/export/download-zip.js]: If server returns a specific error message, show that ..

File: app/components/events/view/export/download-zip.js [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+      .then(res => {
+        const anchor = document.createElement('a');
+        anchor.style.display = 'none';
+        anchor.href = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
+        anchor.download = 'EventExport.zip';
+        anchor.click();
+        this.notify.success(this.l10n.t('Exported Event Downloaded successfully.'));
+      })
+      .catch(e => {
+        console.warn(e);
```
Comment (niranjan94) [inline app/components/events/view/export/download-zip.js]: Should be an error

File: app/components/events/view/export/download-zip.js [hunk 5]
```diff
@@ -1,4 +1,27 @@
...
+  exportEventDownload(eventDownloadUrl) {
+    this.set('isLoading', true);
+    this.loader
+      .load(`${eventDownloadUrl}`, { withoutPrefix: true })
+      .then(res => {
+        const anchor = document.createElement('a');
+        anchor.style.display = 'none';
+        anchor.href = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
+        anchor.download = 'EventExport.zip';
+        anchor.click();
```
Comment (niranjan94) [inline app/components/events/view/export/download-zip.js]: Does this work on all browsers ? Please test on 
- Chrome
- FF
- IE 11
- Edge
- Mobile Chrome
- Mobile Safari 
- Safari

Comment (mrsaicharan1) [reply app/components/events/view/export/download-zip.js]: Tested on Chrome, Safari, Firefox. Works fine.
I've also tested it on Mobile Safari(iPhone) & Mobile Chrome too. Works alright.

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 travis is failing

The loader test seems to be failing. Will see what's wrong & get back

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 travis is failing

The loader test seems to be failing. Will see what's wrong & get back

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 travis is failing

Updated the branch. Travis is passing now.

**Comment by mrsaicharan1** [comment]
> Why is the success message displayed twice in the GIF?

Click on start, on success, message to indicate the exported event is now available for download
Click on download, on success, message to indicate the exported event has been downloaded.

**Pull request merged**